runtime/src/main/java/org/corfudb/runtime/clients/LogUnitClient.java

```diff
@@ -192,6 +211,29 @@ private static Object handleReadDataCorruption(CorfuPayloadMsg<ReadResponse> msg
     }
 
     /**
+     * Asynchronously write a proposal to a logging unit with ranked address space. This proposal data keeps
+     * the rank to the log unit during the two phase recovery write in the quorum replication.
+     *
+     * @param address        The address to write to.
+     * @param streams        The streams, if any, that this write belongs to.
+     * @param rank           The rank of this write]
+     *
+     */
+    CompletableFuture<Boolean> writeProposal(long address, Set<UUID> streams, IMetadata.DataRank rank) {
```

Comment:
In my last client version I changed the signature of this method too and it is now  public CompletableFuture<Boolean> writeEmptyData(long address, DataType type, Set<UUID> streams, IMetadata.DataRank rank)  - we also supply DataType. I will port this and update in the reveiw now (I am in the airport)